File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.Deque;
import java.util.LinkedList;

public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Deque<Integer> q = new LinkedList<>();
        int T = Integer.parseInt(br.readLine());

        for (int i=0; i<T; i++) {
            String[] inputs = br.readLine().split(" ");
            String op = inputs[0];

            if (op.equals("push")) {
                q.add(Integer.parseInt(inputs[1]));
            }

            if (op.equals("pop")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.pop() + "\n");
                }
            }

            if (op.equals("size")) {
                bw.write(q.size() + "\n");
            }

            if (op.equals("empty")) {
                if (q.isEmpty()) {
                    bw.write(1 + "\n");
                } else {
                    bw.write(0 + "\n");
                }
            }

            if (op.equals("front")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.peekFirst() + "\n");
                }
            }

            if (op.equals("back")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.peekLast() + "\n");
                }
            }
        }

        bw.flush();
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

if __name__ == '__main__':

    N = int(sys.stdin.readline())
    q = deque()
    for _ in range(N):

        inputs = sys.stdin.readline().split()
        op = inputs[0]

        if op == "push":
            q.append(int(inputs[1]))

        if op == "pop":
            if len(q) == 0:
                print(-1)
            else:
                print(q.popleft())

        if op == "size":
            print(len(q))

        if op == "empty":
            if len(q) == 0:
                print(1)
            else:
                print(0)

        if op == "front":
            if len(q) == 0:
                print(-1)

            else:
                print(q[0])

        if op == "back":
            if len(q) == 0:
                print(-1)

            else:
                print(q[-1])
